### PR TITLE
Contract update/dx contract rc12

### PR DESCRIPTION
**Pull request by W3stside**
Continued update to dx-contract changes

New additions:
1. integration of `claimAndWithdrawSellerFundsFromSeveral` function newly added
  a. types/tests/prop passing
2. Use of `DutchExchangeHelper` contract for getters used in FE and FE API
3. fixing of some `trufflescripts` for testing ease
4. small `truffle-config` changes/updates

### CODE REVIEW

File: src/actions/blockchain.ts [hunk 1]
```diff
@@ -895,6 +896,58 @@ export const claimSellerFundsFromSeveral = (
...
+      `)
+      dispatch(closeModal())
+      // jump to home Page
+      dispatch(push('/'))
+
+      // grab balance of sold token after claim
+      const balance = await getTokenBalance(sell.address, currentAccount)
+
+      // dispatch Actions
+      dispatch(setTokenBalance({ address: sell.address, balance }))
```
Comment (Velenir) [inline src/actions/blockchain.ts:942]: shouldn't balance getting be outside of `catch {}`?

File: src/api/contracts.ts [hunk 2]
```diff
@@ -25,27 +26,28 @@ const contractNames = [
...
 interface ContractsMap {
-  DutchExchange:  DXAuction,
-  TokenMGN:       MGNInterface,
-  TokenETH?:      ETHInterface,
-  TokenGNO?:      GNOInterface,
-  TokenOWL?:      OWLInterface,
-  TokenOMG?:      GNOInterface,
-  TokenRDN?:      GNOInterface,
+  DutchExchange:        DXAuction,
+  DutchExchangeHelper:  any,
```
Comment (Velenir) [inline src/api/contracts.ts]: You can do here `Pick<DXAuction, functions | you | need>`

File: src/api/index.ts [hunk 3]
```diff
@@ -585,6 +585,70 @@ claimSellerFundsFromSeveralAuctions.sendTransaction = async (
...
+  const finalIndices = claimableIndices.slice(0, length)
+  const sellArr = Array.from({ length }, () => sell.address)
+  const buyArr = Array.from({ length }, () => buy.address)
+
+  console.log('Params = ', sellArr, buyArr, finalIndices, userAccount)
+  return DutchX.claimAndWithdrawTokensFromSeveralAuctionsAsSeller(sellArr, buyArr, finalIndices, userAccount)
+}
+
+claimAndWithdrawSellerFundsFromSeveralAuctions.sendTransaction = async (
+  sell: DefaultTokenObject,
```
Comment (Velenir) [inline src/api/index.ts:620]: a lot of duplication between this function and the one above

File: test/trufflescripts/utils/contracts.js [hunk 4]
```diff
@@ -185,7 +188,7 @@ module.exports = (artifacts) => {
   const giveTokens = (acc, tokensMap, masterAcc) => handleTokensMap(tokensMap, ({ key, token, amount }) => {
     if (key === 'ETH') {
       return token.deposit({ from: acc, value: amount })
-    } else if (masterAcc) {
+    } if (masterAcc) {
```
Comment (Velenir) [inline test/trufflescripts/utils/contracts.js:191]: does this work without a line break?

File: test/trufflescripts/add_token_pair.js [hunk 5]
```diff
@@ -90,16 +93,17 @@ module.exports = async () => {
...
   await dx.addTokenPair(
     sellToken.address,                            // -----> SellToken Address
     buyToken.address,                           // -----> BuyToken Address
     ...funds,                                    // -----> sellFund, buyFund, closingPriceNum, closingPriceDen
     { from: account },
   )
+  await mineCurrentBlock()
   console.log('Auction Index AFTER == ', (await dx.getAuctionIndex.call(sellToken.address, buyToken.address)).toNumber())
-  return mineCurrentBlock()
+  process.exit()
```
Comment (Velenir) [inline test/trufflescripts/add_token_pair.js:108]: does it hang otherwise?

File: src/api/index.ts [hunk 6]
```diff
@@ -585,6 +585,70 @@ claimSellerFundsFromSeveralAuctions.sendTransaction = async (
...
+  buy: DefaultTokenObject,
+  userAccount?: Account,
+  indices: number = 0,
+) => {
+  const { DutchX } = await dxAPI()
+  userAccount = await fillDefaultAccount(userAccount)
+
+  const claimableIndices = (
+    await DutchX.getIndicesWithClaimableTokensForSellers({ sell, buy }, userAccount, indices)
+  )[0].map(i => i.toNumber())
```
Comment (Velenir) [inline src/api/index.ts:599]: I wonder if it would be better to have `.toString()` here, because some years in the future indexes will overflow?

**Review by Velenir — CHANGES_REQUESTED** [review]
This still uses truffle 4 but with solc 0.5, correct?

File: src/actions/blockchain.ts (hunk 1 above)
Comment (W3stside) [reply src/actions/blockchain.ts:942]: no that's a shitty pave over for when web3 acts up but we need to handle state update correctly. I'll look into it more, maybe just a finally block or sth

File: src/api/contracts.ts (hunk 2 above)
Comment (W3stside) [reply src/api/contracts.ts]: :+1:  TS master

File: src/api/index.ts (hunk 3 above)
Comment (W3stside) [reply src/api/index.ts:620]: lol cause it's the `sendTransaction` version, this is a joke right? or am i missing sth?

File: test/trufflescripts/add_token_pair.js (hunk 5 above)
Comment (W3stside) [reply test/trufflescripts/add_token_pair.js:108]: i was testing some stuff, not important. tbh the trufflescripts stuff isn't good, you should just ignore it

File: test/trufflescripts/utils/contracts.js (hunk 4 above)
Comment (W3stside) [reply test/trufflescripts/utils/contracts.js:191]: seems so

File: src/api/index.ts (hunk 6 above)
Comment (W3stside) [reply src/api/index.ts:599]: :+1:

File: src/actions/blockchain.ts (hunk 1 above)
Comment (Velenir) [reply src/actions/blockchain.ts:942]: :+1:

File: src/api/index.ts (hunk 3 above)
Comment (Velenir) [reply src/api/index.ts:620]: I mean L530-549, L561-582, L694-613 and L625-646 are the same.
Would have been a nice joke though :smile:

**Comment by W3stside** [comment]
closing in favour of consolidated branch incoming

**Pull request closed without merging**